single_kernel_mongo/utils/helpers.py

```diff
@@ -0,0 +1,32 @@
+#!/usr/bin/env python3
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Some helpers functions that doesn't belong anywhere else."""
+
+import base64
+import re
+
+
+def parse_tls_file(raw_content: str) -> bytes:
+    """Parse TLS files from both plain text or base64 format."""
+    if re.match(r"(-+(BEGIN|END) [A-Z ]+-+)", raw_content):
+        return (
+            re.sub(
+                r"(-+(BEGIN|END) [A-Z ]+-+)",
+                "\\1",
+                raw_content,
+            )
+            .rstrip()
+            .encode("utf-8")
+        )
+    return base64.b64decode(raw_content)
+
+
+def generate_relation_departed_key(rel_id: int) -> str:  # noqa
+    return f"relation_{rel_id}_departed"
+
+
+def hostname_from_hostport(host: str) -> str:
+    """Takes hostname:port and returns hostname."""
+    return host.split(":")[0]
```

Comment:
shouldn't these also go in their manager?

Reply:
I try to remove complexity of the code by moving out the methods that would only live as classmethod.
Also `generate…` is called in the charm_state not in a manager so either it lives in the charm state (and it doesn't make a big sense) or it leaves in helpers because it's a simple helper onleliner